doc/source/user/absolute_beginners.rst

```diff
@@ -0,0 +1,2464 @@
+
+  Notes
+  -----
+  There are two modes of creating an array using ``__new__``:
+
+  1. If `buffer` is None, then only `shape`, `dtype`, and `order`
+     are used.
+  2. If `buffer` is an object exposing the buffer interface, then
+     all keywords are interpreted.
+
+  No ``__init__`` method is needed because the array is fully initialized
+  after the ``__new__`` method.
+
+  Examples
+  --------
+  These examples illustrate the low-level `ndarray` constructor.  Refer
+  to the `See Also` section above for easier ways of constructing an
+  ndarray.
+
+  First mode, `buffer` is None:
+
+  >>> np.ndarray(shape=(2,2), dtype=float, order='F')
+  array([[ -1.13698227e+002,   4.25087011e-303],
+         [  2.88528414e-306,   3.27025015e-309]])         #random
+
+  Second mode:
+
+  >>> np.ndarray((2,), buffer=np.array([1,2,3]),
+  ...            offset=np.int_().itemsize,
+  ...            dtype=int) # offset = 1*itemsize, i.e. skip first element
+  array([2, 3])
+
+This also works for functions and other objects that **you** create. Just remember to include a docstring with your function using a string literal (``""" """`` or ``''' '''`` around your documentation).
+
+For example, if you create this function:
+
+::
+
+  def double(a):
+    '''Return a * 2'''
+    return a * 2
+
+You can run:
+
+::
+
+  double?
+
+Which will return:
+
+::
+
+  Signature: double(a)
+  Docstring: Return a * 2
+  File:      ~/Desktop/<ipython-input-23-b5adf20be596>
+  Type:      function
+
+You can reach another level of information by reading the source code of the object you're interested in. Using a double question mark (``??``) allows you to access the source code.
+
+For example, running:
+
+::
+
+  double??
+
+Will return 
+
+::
+
+  Signature: double(a)
+  Source:   
+  def double(a):
+      '''Return a * 2'''
+      return a * 2
+  File:      ~/Desktop/<ipython-input-23-b5adf20be596>
+  Type:      function
+
+If the object in question is compiled in a language other than Python, using ``??`` will return the same information as ``?``. You'll find this with a lot of built-in objects and types, for example:
+
+::
+
+  len?
+
+**Output:**
+
+::
+
+  Signature: len(obj, /)
+  Docstring: Return the number of items in a container.
+  Type:      builtin_function_or_method
+
+and
+
+::
+
+  len??
+
+**Output:**
+
+::
+
+  ​Signature: len(obj, /)
+  Docstring: Return the number of items in a container.
+  Type:      builtin_function_or_method
+
+have the same output because they were compiled in a programming language other than Python.
+
+
+
+Working with mathematical formulas
+----------------------------------
+
+Implementing mathematical formulas that work on arrays is one of the things that make NumPy so highly regarded in the scientific Python community. 
+
+For example, this is the mean square error formula (a central formula used in supervised machine learning models that deal with regression):
+
+.. image:: images/np_MSE_formula.png
```

Comment:
This would be better as LaTeX. You can use color in mathjax as `{ \color{green} \frac{1}{n} }`, or possibly `\color{green}{\frac{1}{n}}`, depending on which version we're using.

Reply:
I'm not sure how to get LaTex/MathJax to correctly render in .rst/Sphinx. Could you please provide a little more direction?

Reply:
Wrapping as:
```
.. math::

    { \color{green} \frac{1}{n} } rest of the equation
```
should do the trick

Reply:
Like this?
.. math::

    MeanSquareError = { \color{green} \frac{1}{n} }
    { \color{blue} \sum{i=1}^{n} } 
    { \color{red} (predictions - }
    { \color{purple} labels) }
    { \color{yellow} ^{2} }

Reply:
The math formula illustration looks like:

<img width="560" alt="image" src="https://user-images.githubusercontent.com/98330/69829734-174fca00-11d6-11ea-874d-5bbd6d6430bc.png">

@eric-wieser suggested to change to MathJax (not sure only for the first formula, or for more?). The colors can be replicated that way, but the visualization of the arrays inside the formulas can't. Given the "absolute beginner" audience, I think this looks quite friendly. The first math formula does look the worst of all these images though, so worth a try to see if MathJax makes it looks better.

Reply:
@bonn0062 , I think you'll want to put the parentheses and minus sign outside the colored sections, else the spacing will end up wrong

Reply:
The images look great, but by showing the arrays vertically I fear we're only adding to the 1d vs 2d confusion faced by former Matlab users. If nothing else, the arrays appear horizontal in output yet vertically here.

Reply:
> depending on which version we're using.

Looks like v3 isn't actually release yet, so you probably need the `\color{green}{contents}` syntax instead.

Reply:
.. math::

    MeanSquareError = \color{green}{\frac{1}{n}}
    \color{blue}{\sum{i=1}^{n}} (
    \color{red}{predictions} - 
    \color{purple}{labels} )
    \color{yellow}{^{2}}

Reply:
Does this look right for syntax?

Reply:
Well, that's definitely not correct. Circleci was horrified :)

Reply:
Do you have the link to the failed circleCI run??

That's mostly ok, but the `^` needs to be outside the coloring. Perhaps:
```
.. math::

    MeanSquareError = \color{green}{\frac{1}{n}}
    \color{blue}{\sum{i=1}^{n}} (
    \color{red}{predictions} - 
    \color{purple}{labels} )
    ^{{\color{yellow}{2}}
```

You might also want to wrap all of the text with `\mathrm{}` or `\texttt{}` to make it not look like a product of single-letter variables.

Reply:
It's here! https://circleci.com/gh/numpy/numpy/10389?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link Do you think this will work without MathJax? https://github.com/numpy/numpy/pull/14546#issuecomment-559883524

Reply:
For sure it should be possible to get this to work - I think you need to:

* set `imgmath_latex_preamble = R"\usepackage{xcolor}"` in conf.py (or append to any existing value)
* replace `\color` with `\textcolor` above

In the long run, I'd consider switching to mathjax, but that's out of scope for this patch.

Reply:
Hmmmmm...my guess is it would go here https://github.com/bonn0062/numpy/blob/absolute-beginners/doc/source/conf.py just after line 180 or so, but I'm thinking I won't mess with conf.py unless I'm absolutely sure that's the right thing to do and exactly what to do :)

Reply:
Yeah, better not to touch that in this PR. It'd be a change like https://github.com/scipy/scipy/pull/7376

Reply:
To be clear @rgommers, you're fine with a preamble change, just not the change to mathjax

@bonn0062, that sounds right to me - please go ahead and do so! Line 183 looks like a reasonable place to me, just after the other usepackage.

Reply:
Preamble sounds fine. I think MathJax could be done too, but better in a separate PR and then put this one on top.

Reply:
I added that line exactly as written above to line 183 on a separate pull request (just in case I break something with this...). If you want it on this pr, just say the word. https://github.com/numpy/numpy/pull/15016